BLSQ/iaso
# IA-2188: users  /  projects & bulk update

Link user profile to projects.
Possibility to add or remove project to a specific user.
Allow to filter user by project
Bulk update of user to change roles, project, language, locations

Related JIRA tickets : IA-2188

## Self proofreading checklist


## Changes

- refactor user dialog to use MakeFullModal
- validate data before saving (username and password required)
- adding project migration to users
- modify api to display project and filter by projects
- modify save of user to include project
- new bulk update query, dialog, and multi selection
- bulk query endpoint
- bulk task to update multiple profiles
- front-end and back-end tests


## How to test

- Go to user page
- try to add and remove project to an user
- play with the multi selection and launch a bulk update (worker needs to be on)
- check that bulk update modifications are ok

## Print screen / video


https://github.com/BLSQ/iaso/assets/12494624/1ff14ab4-a534-4a92-9a9a-9a7b56b940d0
